CONTRIBUTING.md

```diff
@@ -7,39 +7,91 @@ out of applications and platforms. Many also add a personal perspective to the
+
+- Have an active GitHub account
+- Install Hugo on your local machine
+  -  ```make install hugo```
+
+## Prepare to contribute
+
+We encourage a triangular workflow in GitHub as a best practice.
 
 1. Fork this repo (https://github.com/rackerlabs/technical-blog), then
    clone your fork.
 
-2. Before you start working, make sure your local content is up-to-date and
-   merged with the right upstream branch:
+   ```bash
+   git clone git@github.com:<YOUR_GITHUB_ID>/technical-blog.git
+   cd technical-blog
+
+1. Add the original repo as the `upstream` remote.
+
+   ```bash
+   git remote add upstream git@github.com:rackerlabs/technical-blog.git
+   git remote update
+   ```
+
+1. Configure Git to always push to origin (your forked copy) using the current branch name.
+
+   ```bash
+   git config remote.pushdefault origin
+   git config push.default current
+   ```
+
+1. Prevent accidental commits directly to the master branch by using a pre-commit hook.
+   Note: These instructions work in the Bash shell and will overwrite the existing file if present.
+
+   ```bash
+   mkdir -p ~/.git/hooks
+   cat << 'EOF' > ~/.git/hooks/pre-commit
+   #!/usr/bin/env bash
+
+   current_branch=$(git symbolic-ref -q HEAD | sed -e 's|^refs/heads/||')
+
+   if [[ $current_branch = 'master' ]]; then
+      echo 'Direct commits to the master branch are not allowed.'
+      exit 1
+   fi
+   EOF
+   ```
+
+1. Set your local master branch to track `upstream/master`.
+
+   ```bash
+   git branch master -u upstream/master
+   ```
+
+## Create your blog post
+
+1. Always keep your fork up to date (rebase) before starting work.
 
       ```bash
-         git remote add upstream git@github.com:rackerlabs/docs-developer-blog.git
-         git checkout master
-         git fetch upstream
-         git merge upstream/master
-         git checkout -b name-of-your-branch
-      ```
+      git checkout master
+      git fetch --all
+      git rebase upstream/master
+      git push origin master
 
-3. Create a new folder inside the `/hugo/content/blog` directory with the following naming
-   convention: `YYYY-MM-DD-title-of-your-post`, where `YYYY-MM-DD` is the
-   date you want your entry to be published.
+1. Create a new branch for your blog.
 
-4. Within the newly created folder, create a new file with the following naming convention:
-   `index.md`.The published URL for your post becomes `https://docs.rackspace.com/blog/title-of-your-post/` when published.
+      ```bash
+      git checkout -b name-of-your-branch
+      ```
+1. Navigate to the root directory of your local copy of this repo using the command line.
 
-**Note:** If your post has images, you should place the image files within the newly created
-folder.
+1. Create a new directory within ```/content/blog/YYYY/``` that contains the inital ```.index.md``` 
+   by using the following command:
+  
+   ```hugo new --kind blog-bundle blog/YYYY-MM-DD-title-of-post```
```

Comment:
@WilliamLoy please change this to hugo new --kind blog-bundle blog/YYYY/YYYY-MM-DD-title-of-post